kie-plugins-testing/src/test/java/org/kie/maven/plugin/CompilerHelperTest.java

```diff
@@ -0,0 +1,500 @@
+import org.kie.api.conf.EqualityBehaviorOption;
+import org.kie.api.conf.EventProcessingOption;
+import org.kie.api.definition.type.Role;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.StatelessKieSession;
+import org.kie.api.runtime.conf.ClockTypeOption;
+
+import static ch.qos.logback.core.encoder.ByteArrayUtil.hexStringToByteArray;
+import static ch.qos.logback.core.encoder.ByteArrayUtil.toHexString;
+import static org.assertj.core.api.Assertions.*;
+
+public class CompilerHelperTest {
+
+    private static Log log;
+    protected FileManager fileManager;
+
+    @Before
+    public void setUp() {
+        log = new SystemStreamLog();
+        this.fileManager = new FileManager();
+        this.fileManager.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        this.fileManager.tearDown();
+    }
+
+    @Test
+    public void getCompilationIDTest() {
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> map = new HashMap<>();
+        String uuid = UUID.randomUUID().toString();
+        map.put("compilation.ID",
+                uuid);
+        String compilationID = helper.getCompilationID(map,
+                                                       log);
+        assertThat(compilationID).isEqualTo(uuid);
+    }
+
+    @Test
+    public void getWrongCompilationIDTest() {
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> map = new HashMap<>();
+        String uuid = UUID.randomUUID().toString();
+        map.put("compilation..ID",
+                uuid);
+        String compilationID = helper.getCompilationID(map,
+                                                       log);
+        assertThat(compilationID).isEqualTo("main");
+    }
+
+    @Test
+    public void shareStoreWithMapTest() {
+        //prepare
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> kieMap = new HashMap<>();
+        String compilationID = UUID.randomUUID().toString();
+        kieMap.put("compilation.ID",
+                   compilationID);
+        ProjectClassLoader pcl = ProjectClassLoader.createProjectClassLoader();
+        String hexContent = "e04fd020ea3a6910a2d808002b30309d";
+        String className = "org/my/Clazz";
+        //we store bytes of hexString for an easy compare
+        pcl.storeClass(className,
+                       hexStringToByteArray(hexContent));
+
+        //call the class under test
+        helper.shareStoreWithMap(pcl,
+                                 compilationID,
+                                 kieMap,
+                                 log);
+
+        //verify
+        StringBuilder sbTypes = new StringBuilder(compilationID).append(".").append("ProjectClassloaderStore");
+        kieMap.get(sbTypes.toString());
+        Map store = (Map) kieMap.get(sbTypes.toString());
+        assertThat(store).isNotNull();
+        assertThat(store.size()).isEqualTo(1);
```

Comment:
Can be used `assertThat(store).hasSize(1);`

Reply:
Fixed